android/src/main/java/com/polidea/reactnativeble/BleModule.java

```diff
@@ -1429,17 +1850,24 @@ private void cleanServicesAndCharacteristicsForDevice(@NonNull Device device) {
             int key = discoveredServices.keyAt(i);
             Service service = discoveredServices.get(key);
 
-            if (service.getDevice().getNativeDevice().getMacAddress().equals(device.getNativeDevice().getMacAddress())) {
+            if (service.getDeviceId().equals(device.getDeviceId())) {
                 discoveredServices.remove(key);
             }
         }
         for (int i = discoveredCharacteristics.size() - 1; i >=0; i--) {
             int key = discoveredCharacteristics.keyAt(i);
             Characteristic characteristic = discoveredCharacteristics.get(key);
 
-            if (characteristic.getService().getDevice().getNativeDevice().getMacAddress().equals(device.getNativeDevice().getMacAddress())) {
+            if (characteristic.getDeviceId().equals(device.getDeviceId())) {
```

Comment:
Some of the changes seem not related to the topic of the PR. Just sayin'

Reply:
I agree I need to pick a different strategy for pull requests. What I want to achieve is consistent `master` branch at all times. Probably next time I will just send smaller PR to a different branch and merge it to `master` afterward.